ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/HiveMetaStoreAuthorizer.java

```diff
@@ -108,7 +110,10 @@ public final void onEvent(PreEventContext preEventContext)
       }
     } catch (Exception e) {
       LOG.error("HiveMetaStoreAuthorizer.onEvent(): failed", e);
-      throw new MetaException(e.getMessage());
+      MetaException me = new MetaException(e.getMessage());
+      // Init cause for testing check.
+      me.initCause(e);
```

Comment:
Should not be better extend the MetaException? As I remember it can receive an object already.  It is just style. I personally do not like the initCause :)

Reply:
Constructor `public MetaException(MetaException other)` will only set message, so we can not get cause if we do not call initCause. BTW the causes of MetaException will not be transported to client side, the most valuable function to init cause is for RetryingHMSHandler check and root cause check in server side IMHO.

Reply:
@wecharyu please check `MetaStoreUtils.newMetaException(Exception e)`
````
public static MetaException newMetaException(String errorMessage, Exception e) {
    MetaException metaException = new MetaException(errorMessage);
    if (e != null) {
      metaException.initCause(e);
    }
    return metaException;
  }
````

Reply:
````
throw MetaStoreUtils.newMetaException(e)
````

Reply:
Addressed.